### PR TITLE
World select

### PR DESCRIPTION
**Pull request by stallop**
World select (hopefully) working.

Note: To test all the functionalities, I have the save data currently put one level in world 2, which is unlocked, and two levels in world 3, which are both locked. I've been trying different combinations to make sure unlocking worlds, unlocking levels, and pagination all work together properly, but might have missed some. Feel free to mess with the save json by moving levels around to different worlds and locking/unlocking them to see if there's any cases the code doesn't properly handle.

### CODE REVIEW

File: core/assets/savedata/save.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
+[
+{
```
Comment (BryanGraeser) [inline core/assets/savedata/save.json]: This file shouldn't exist (delete this)

**Review by BryanGraeser — CHANGES_REQUESTED** [review]
This is looking like a good start, but there are a number of issues I noticed:
- If you unlock a new world while playing, this is not reflected in the map until you close & quit (ex: I unlocked the last world and exited to the menu to view the map and it was not unlocked)
- There should be something on the level select screen reflecting what world you are in, otherwise it is confusing. Perhaps even just changing the title (ex: "Forest Level Select" vs "Cave Level Select").  Perhaps a different background too (like just tinting the same background image to green for forest, and red for volcano. You may want to talk to Ravina about this one.
- Also this may be more a loading screen PR issue, but finishing the last level in a world should send you back to the map (so you know you beat the world), not straight on to the next level
- Very minor thing, but the "back" button on the level select screen should say probably "World Select" or "Back to World Select" now
- Finally, there are multiple pages in Level Select when there shouldn't be (looks like it is basing the # of pages off of total levels, not levels in that world). Screenshots included below.


image

File: core/assets/savedata/save.json (hunk 1 above)
Comment (BryanGraeser) [reply core/assets/savedata/save.json]: The whole file (just commented a couple lines)

**Comment by BryanGraeser** [comment]
Also note for anyone testing this: delete the savedata folder in your $HOME directory before testing (this will create a new save.json based off of the template one which Max has updated). This is necessary because obviously we can't push changes to files in your $HOME directory to this git repo.

**Also note you will need to do the same again when going back to another branch (until this is merged).**

**Comment by stallop** [comment]
> This is looking like a good start, but there are a number of issues I noticed:
> 
> * If you unlock a new world while playing, this is not reflected in the map until you close & quit (ex: I unlocked the last world and exited to the menu to view the map and it was not unlocked)
> * There should be something on the level select screen reflecting what world you are in, otherwise it is confusing. Perhaps even just changing the title (ex: "Forest Level Select" vs "Cave Level Select").  Perhaps a different background too (like just tinting the same background image to green for forest, and red for volcano. You may want to talk to Ravina about this one.
> * Also this may be more a loading screen PR issue, but finishing the last level in a world should send you back to the map (so you know you beat the world), not straight on to the next level
> * Very minor thing, but the "back" button on the level select screen should say probably "World Select" or "Back to World Select" now
> * Finally, there are multiple pages in Level Select when there shouldn't be (looks like it is basing the # of pages off of total levels, not levels in that world). Screenshots included below.
>   
> 
> image

all of these should now be fixed.

**Review by lsizemore8 — CHANGES_REQUESTED** [review]
This is a lot better! Two small fixes that I noticed left: 
1. When you win the game, it should also go back to the world select
2. This is actually a LevelSelectMode issue. When I hover over a level that is not unlocked, but does exit, it does the hover effect, even though it's not unlocked.

**Comment by lsizemore8** [comment]
Last bug, I hope. If you do the following chain of events: 
World Select -> Cave -> Page 2 -> back to World Select -> Forest, the forest level select page will start on an empty page (ie its page 2, which technically doesn't exist)

File: core/src/com/fallenflame/game/GDXRoot.java [hunk 2]
```diff
@@ -167,15 +172,29 @@ else if(loading.toCredits){
...
-					setScreen(levelSelect);
+				} else {
+					worldSelect.reset(engine.getLevelSaves());
+					Gdx.input.setInputProcessor(worldSelect);
+					worldSelect.setScreenListener(this);
+					setScreen(worldSelect);
 				}
 			}
 
 //			loading.dispose();
```
Comment (lsizemore8) [inline core/src/com/fallenflame/game/GDXRoot.java:183]: Remove the comments

**Review by lsizemore8 — APPROVED** [review]
Merge in master/remove commented out code if we don't need it, but it looks good to me

**Comment by BryanGraeser** [comment]
One very small thing: I think it should say "World Map", not "Map" on the top

**Comment by BryanGraeser** [comment]
But other than that I think it's good to go!

**Review by BryanGraeser — APPROVED** [review]

**Pull request merged**